pkg/cmd/controller/csr.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"context"
+	"github.com/openshift/cluster-policy-controller/pkg/operator"
+	"github.com/openshift/library-go/pkg/operator/csr"
+	"github.com/openshift/library-go/pkg/operator/events"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+const (
+	controllerName = "csr-approver-controller"
+	// Hardcoded for the Monitoring Operator:
+	// https://github.com/openshift/cluster-monitoring-operator/blob/master/manifests/0000_50_cluster-monitoring-operator_03-role-binding.yaml#L25-L27
+	monitoringServiceAccountNamespace = "openshift-monitoring"
+	monitoringServiceAccountName      = "cluster-monitoring-operator"
+	monitoringCertificateSubject      = "/CN=prometheus"
+)
+
+func RunCSRApproverController(ctx *ControllerContext) (bool, error) {
+	kubeClient, err := ctx.ClientBuilder.Client(infraClusterCSRApproverControllerServiceAccountName)
+	if err != nil {
+		return true, err
+	}
+
+	operatorClient := &operator.OperatorClient{}
```

Comment:
we'll want to get https://github.com/openshift/library-go/pull/1105 merged first to use it here

Reply:
I proposed slightly different (more async) approach to this here: https://github.com/openshift/cluster-policy-controller/pull/61#issuecomment-863886456